src/sql/src/query_model/mod.rs

```diff
@@ -23,8 +23,37 @@ mod test;
 
 pub use scalar_expr::*;
 
-pub type QuantifierId = u64;
-pub type BoxId = u64;
+#[derive(Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct QuantifierId(u64);
+#[derive(Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct BoxId(u64);
+
+impl std::fmt::Display for QuantifierId {
```

Comment:
Is `Display` actually used? If not I'd prefer to only have the `Debug` implementation. It's less code to maintain :)

Reply:
I did this this in 76560b17 to illustrate where the changes are.

My preference would be to leave it as is. Some of the client code is `dot.rs` where the formatted output is used to construct node names for the dot file, which strictly speaking is not "for debugging purposes".

Reply:
As an alternative, implement `Display` for `Gen` that internally uses `Debug`. But as you prefer, it's no big thing!